test/unit/puppetlabs/services/ca/certificate_authority_core_test.clj

```diff
@@ -407,13 +416,23 @@
             (is (= #{localhost-status test-agent-status revoked-agent-status}
                  (set (json/parse-string (:body response) true))))))
 
+        (testing "returns json when no accept header specified"
+          (let [response (test-app
+                          {:uri "/v1/certificate_statuses/thisisirrelevant"
+                           :request-method :get
+                           :headers {}})]
+            (is (= 200 (:status response)))
+            (is (.startsWith (get-in response [:headers "Content-Type"]) "application/json"))
+            (is (= #{localhost-status test-agent-status revoked-agent-status}
+                   (set (json/parse-string (:body response) true))))))
+
         (testing "with 'Accept: pson'"
           (let [response (test-app
                           {:uri "/v1/certificate_statuses/thisisirrelevant"
                            :request-method :get
                            :headers {"accept" "pson"}})]
           (is (= 200 (:status response)))
-          (is (.startsWith (get-in response [:headers "Content-Type"]) "text/pson"))
+          (is (.startsWith (get-in response [:headers "Content-Type"]) "pson"))
```

Comment:
No, this would be a result of the changes on my PR.  Previously, it was defaulting to `text/pson` because liberator was failing to pass through the value from the `accept` header as the `media-type` for the representation - probably because it thinks that `pson` looks like a bogus content-type - and so we were falling back to delivering our hard-coded internal default.  I think the original implementation was actually buggy but `text/pson` and `pson` are somewhat interchangeable so we probably just accepted this at the time.

When I initially tried just changing that default to `application/json`, it resulted in an `accept` header of `pson` for the request returning a `content-type` of `application/json` in the response, which seemed wrong.  For the changes on the PR, then, I just changed the implementation to try to roundtrip the value from the `accept` header in the event that there is one and only fallback to `application/json` in the event that the client didn't provide an `accept` header.  That seems like the best thing to do from a standards perspective.  It would seem that a legacy client which sent an `accept` header of `pson` should know how to deal with a `Content-Type` of `pson` in the response.

Does this make sense?